packages/react-components/src/components/chart/chartOptions/seriesAndYAxis/convertSeriesAndYAxis.ts

```diff
@@ -216,38 +226,92 @@ export const reduceSeriesAndYAxis = (
  * @returns { series, yAxis } converted series options and converted yAxis options
  */
 export const useSeriesAndYAxis = (
-  datastreams: DataStream[],
+  datastreams: DataStreamInfo[],
   {
+    defaultVisualizationType,
+    significantDigits,
     styleSettings,
     axis,
     thresholds,
     performanceMode,
-  }: {
-    styleSettings: StyleSettingsMap;
+  }: Pick<
+    ChartOptions,
+    'defaultVisualizationType' | 'styleSettings' | 'significantDigits'
+  > & {
     thresholds: Threshold[];
     axis?: ChartAxisOptions;
     performanceMode?: boolean;
   }
 ) => {
-  const defaultYAxis: YAXisComponentOption[] = [convertChartYAxis(axis)];
-  const convertedThresholds = convertThresholds(thresholds);
-  const getStyles = getChartStyleSettingsFromMap(styleSettings);
-
-  const { series, yAxis } = datastreams
-    .map((datastream) =>
-      convertSeriesAndYAxis(getStyles(datastream), { performanceMode })(
-        datastream
-      )
-    )
-    .reduce(reduceSeriesAndYAxis, { series: [], yAxis: defaultYAxis });
-
-  if (series.length > 0) {
-    series[0].markArea = convertedThresholds.markArea;
-    series[0].markLine = convertedThresholds.markLine;
-  }
+  const { highlightedDataStreams } = useChartStore(
+    (state) => ({ highlightedDataStreams: state.highlightedDataStreams }),
+    isEqual
+  );
+  const isDataStreamHighlighted = useCallback(
+    (datastream: DataStreamInfo) =>
+      isDataStreamInList(highlightedDataStreams)(datastream),
```

Comment:
its probably worth to update isDataStreamInList to just accept datastream as a prop and not return a function which accepts datastram

Reply:
any thoughts ?

Reply:
Can we address this change in a separate PR? That change will also affect the legend table, and this PR is already quite big.